addons/project/project.py

```diff
@@ -105,9 +105,12 @@ def _get_attached_docs(self, cr, uid, ids, field_name, arg, context):
             res[id] = (project_attachments or 0) + (task_attachments or 0)
         return res
     def _task_count(self, cr, uid, ids, field_name, arg, context=None):
+        if context is None:
+            context = {}
         res={}
-        for tasks in self.browse(cr, uid, ids, dict(context, active_test=False)):
-            res[tasks.id] = len(tasks.task_ids)
+        for project in self.browse(cr, uid, ids, context=context):
+            tasks_undefined_stage = self.pool['project.task'].search_count(cr, uid, [('stage_id', '=', None), ('project_id', '=', project.id)], context=context)
+            res[project.id] = len(project.task_ids) + int(tasks_undefined_stage) 
```

Comment:
Currently a project doesn't have stages by default. You may have a task for this project in 'Undefined' stage, which is not a stage at all. This task is not indexed in project.task_ids, and the counter will display 0. 

Also, "for tasks in self.browse()" was weird, as the self is a project.